### ISSUE TITLE
FLASK/DJANGO plugins ENVVAR_PREFIX must accept lowercase for consistency - Docker compose set environment variables vs .toml files

### ISSUE THREAD

**Issue by lowercase00** [rochacbruno/dynaconf#719]
Hi all.

We are just making the jump from .env files to .toml files with Dynaconf for a Flask app.
One thing that we used to do was to have a DB_URI in the .env file `http://localhost:5432/mydb` and another in the docker-compose file, that referenced the local db (outside the container), like `http://docker.host.internal:5432/mydb`.
That would allow us to make migrations (`flask migrate`), that would use the .env set DB_URI, and also to run the docker-compose (since the env var set on the compose file would overwrite the env var set on the .env file).

Using Dynaconf, we noticed that variables are not being loaded from the compose file, it seems that the .toml files have priority, and so the compose file env vars end up being ignored. The work around is to keep changing the variables in the toml file.

I imagine this is fairly specific to our development environment (DB outside the container ecosystem), but I was wondering if this is the expected behavior. We could probably have different environments stacked on the toml files, adding a clone of the "local" environment, creating a new "docker" environment. This would probably work, but just was curious about the variables precedence logic.

Amazing lib by the way, thanks a lot for sharing!
Cheers

**Comment by rochacbruno** [comment]
Hi @lowercase00 

Environment variables have the higher precedence and must override anything on the toml files.
what happens is that Dynaconf assumes there is a prefix for every env var.

By default the prefix is `DYNACONF_` so `DB_URI` becomes `DYNACONF_DB_URI` for regular Python apps and `FLASK_DB_URI` for the FlaskDynaconf object.

You can customize that prefix on your settings object.

Python

```py
settings = Dynaconf(envvar_prefix="MYAPP")
``` 

Flask

```py
settings = FlaskDynaconf(app, ENVVAR_PREFIX="MY_APP")
```

If you set `envvar_prefix` variable to `False` then you get dynaconf to load all environment variables with no prefix needed.

See more on the docs https://www.dynaconf.com/envvars/#custom-prefix

If that is not working for you, then I can help or if you found a bug we can label this issue as a bug.

**Comment by lowercase00** [comment]
Hey @rochacbruno thanks a lot for the fast response!

So, I'm not too certain I understood the flow correctly, but as I understand, I should be able to set the custom prefix:

```python
...
FlaskDynaconf(app, envvar_prefix="PEANUT", settings_files=['.secrets.toml")
```

I would then use
```toml
# .secrets.toml
MY_VAR = "my custom variable" 
```

And then in `docker-compose` I would do:
```yml
version: "3.7"
services:
  peanut_app:
    build: .
    restart: always
    container_name: peanut_app
    environment:
      - PEANUT_MY_VAR=database_address_from_docker_env
    ports:
      - "5000:5000"
```

Does that make sense? I imagine that if I ran the application with Flask directly I would get `MY_VAR=set_by_flask`, and if I ran with `docker-compose`  I would get: `MY_VAR=database_address_from_docker_env`.

I made a [repo](https://github.com/lowercase00/dynaconf_test) to isolate my current project environment, I could replicate the challenge I was having, and I can't seem to find a way to have the variable that was set in the compose file.

Let me know if my understanding is correct. If that's the case, I can dig deeper to try to understand the cause of this specific behavior. 

Cheers!

**Comment by lowercase00** [comment]
@rochacbruno 
Adding to the previous posts, I've done some further testing, both locally and in a Kubernetes.
The test is something like this:

Local:
```python
# app.py
from flask import Flask, jsonify
from dynaconf import FlaskDynaconf

app = Flask(__name__)
FlaskDynaconf(app=app, envvar_prefix="PEANUT")

@app.route('/')
def main():
    return jsonify(var=app.config['APP_VAR'])
```
Then I make three runs:

```bash
# Run 1
## Terminal 1
export FLASK_APP_VAR="from flask"
export FLASK_APP=app.py
flask run

## Terminal 2
curl localhost:5000
> {"var": "from flask"}
```

```bash
# Run 2
## Terminal 1
export DYNACONF_APP_VAR="from dynaconf"
export FLASK_APP=app.py
flask run

## Terminal 2
curl localhost:5000
> {"var": "from flask"}
```

```bash
# Run 3
## Terminal 1
export PEANUT_APP_VAR="from peanut"
export FLASK_APP=app.py
flask run

## Terminal 2
curl localhost:5000
> KeyError: 'APP_VAR'
```


```bash
# Run 4
## Change `envvar_prefix` to uppercase: `ENVVAR_PREFIX`
## Terminal 1
export PEANUT_APP_VAR="from peanut"
export FLASK_APP=app.py
flask run

## Terminal 2
curl localhost:5000
> {"var": "from peanut"}
```

I could replicate the same behavior on Kubernetes using a ConfigMap and a Deployment `enfFrom: ConfigMap`.
It seems that `FlaskDynaconf` is case sensitive for the arguments (accepts ENVVAR_PREFIX) while Dynaconf is case insensitive for the envvar_prefix variable.

As I suggestion, it could be that case to make the behavior consistent across the plugin (both sensitive or both insensitive. To avoid breaking changes, it could be possible to add both uppercase and lowercase arguments to the class OR just add `.upper()` to the variable. The other option would be to just change the documentation.

I would be more than happy to help for any scenario, let me know if you think those tests make sense.
Cheers!

**Comment by rochacbruno** [comment]
@lowercase00 I changed the title of this issue and labeled as a bug.

Thanks for the investigation

**Comment by lowercase00** [comment]
@rochacbruno sure, my pleasure. This if my first PR in here, let me know if any other modification is needed. Also tried to improve the docs in that sense as well.

**Comment by rochacbruno** [comment]
fixed on #721
